src/_pytest/config/__init__.py

```diff
@@ -1351,6 +1356,27 @@ def getvalueorskip(self, name: str, path=None):
         """ (deprecated, use getoption(skip=True)) """
         return self.getoption(name, skip=True)
 
+    def _warn_about_missing_assertion(self, mode: str) -> None:
+        if not _assertion_supported():
+            from _pytest.warnings import _issue_warning_captured
+
+            warning_text = (
+                "assertions not in test modules or"
+                " plugins will be ignored"
+                " because assert statements are not executed "
+                "by the underlying Python interpreter "
+                "(are you using python -O?)\n"
+            )
+            if mode == "plain":
+                warning_text = (
+                    "ASSERTIONS ARE NOT EXECUTED"
+                    " and FAILING TESTS WILL PASS.  Are you"
+                    " using python -O?"
+                )
+            _issue_warning_captured(
+                PytestConfigWarning(warning_text), self.hook, stacklevel=2,
```

Comment:
Same question here re. stacklevel.

Reply:
same reply as to your original question. That would mean we want stacklevel 2 here as well right?